argo-workflows/sync-nb-server-to-ironic/workflowtemplates/sync-server-to-ironic.yaml

```diff
@@ -11,18 +11,28 @@ spec:
       - name: interface_update_event
         value: "{}"
   templates:
-    - name: sync-server
-      container:
-        image: ghcr.io/rackerlabs/understack/ironic-nautobot-client:latest
-        command:
-          - sync-server
-        args:
-          - "{{workflow.parameters.interface_update_event}}"
-        volumeMounts:
-          - mountPath: /etc/ironic-secrets/
-            name: ironic-secrets
-            readOnly: true
-      volumes:
-        - name: ironic-secrets
-          secret:
-            secretName: production-ironic-for-argo-creds
+    - name: main
+      input:
+        parameters:
+          - name: interface_update_event
+      output:
+        parameters:
+          - name: server_state
+            valueFrom:
+              expression: "{{ steps.sync-server.outputs.parameters.result }}"
+      steps:
+        - - name: sync-server
```

Comment:
Why did you make this to be a nested workflow instead of just putting the input and output parameters on the existing step? I'm looking at https://argo-workflows.readthedocs.io/en/latest/walk-through/output-parameters/

Reply:
I was just trying to follow how other workflows were setup but made changes accordingly please review and let me know if any other changes are needed

Reply:
So this converts this from a single step workflow (a single drop in function) into a chain of steps workflow with just 1 step. This means we'll get the overhead of a workflow runner instead of just the one step. I would revert this back.

In fact this change will make it that the output parameters won't work. Because you're just returning the output parameter into the chain of steps and not out of the actual workflow itself.

Reply:
https://github.com/argoproj/argo-workflows/blob/main/examples/output-parameter.yaml so that's from their docs for example.

Reply:
ok got it, will revert it back to the nested workflow that I had

Reply:
I have used the text file generated by the script and returning the output from the workflow in single step.